systemtest/elasticsearch.go

```diff
@@ -90,6 +91,19 @@ func newElasticsearchConfig() elasticsearch.Config {
 // and ingest node pipelines whose names start with "apm",
 // and deletes the default ILM policy "apm-rollover-30-days".
 func CleanupElasticsearch(t testing.TB) {
+	for {
+		err := tryCleanupElasticsearch(t)
+		if err == nil {
+			return
+		}
+		// Retry deleting, in case indices are still being deleted.
+		const delay = 100 * time.Millisecond
+		t.Logf("failed to cleanup Elasticsearch resources (retrying in %s): %s", delay, err)
+		time.Sleep(delay)
+	}
```

Comment:
I agree with your comment in https://github.com/elastic/apm-server/issues/4366#issuecomment-728676137 that we should avoid pure retrying when possible. 
My experience from retrying in the python system test is similar, issues with poorly written tests are hidden for some time, but come back as flakyness out of the blue. Therefore if you don't mind, I'd prefer calling the `_refresh` API instead of retrying.

Reply:
OK, I'll create a new PR and close this. We really should get to the bottom of the cause, thanks for challenging me on this.